### PR TITLE
SCUMM: MI2: Kill hanging tunes on start of next, Fixes bug #1410

### PR DESCRIPTION
**Pull request by BenCastricum**
Fix the "MI2: Two soundtracks playing at once" bug from the forum, as
well as the one the bug reporter experienced.

Both are caused by starting tunes as trigger from other tunes. This code suppresses the start or kills the hanging tune on start of the next. I left in some debug statements to see when the code kicks in so we can determine if there are any unexpected side effects. In the final version those probably should be removed.

### CODE REVIEW

**Comment by BenCastricum** [comment]
The debugging code has been removed. I think it's merge-able now.

File: engines/scumm/imuse/imuse.cpp [hunk 1]
```diff
@@ -675,6 +675,44 @@ bool IMuseInternal::startSound_internal(int sound, int offset) {
...
+	// 101 and 101 moves to an outro and stop (triggering start of 107).
+	// Then at Captain Dread tune 107 is stopped and 100 is started.
+	//
+	// If you go quickly enough, the trigger occurs not at the Scabb Island
+	// map but at Captain Dread causing tune 107 not to be stopped.
+	// So we prevent starting 107 if 100 is already running.
+	if (_game_id == GID_MONKEY2 && (sound == 107) && (getSoundStatus_internal(100, true) == 1))
+		return false;
+
+	/* In some cases 107 is running and doesn't get killed at Dread's map */
```
Comment (bluegr) [inline engines/scumm/imuse/imuse.cpp]: Why do you change comments to use /* .. */ here, when you're using // in the rest of your changes?

Comment (BenCastricum) [inline engines/scumm/imuse/imuse.cpp]: Just an old habbit from other projects. I fixed it.

**Comment by digitall** [comment]
@sev- : This seems fairly reasonable and self-contained. Any objections to just merging this?

**Comment by sev-** [comment]
Right, merging

**Pull request merged**